api/rql/ast/metaPrimaryRealWorld_test.go

```diff
@@ -0,0 +1,170 @@
+package ast
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"testing"
+
+	"github.com/puppetlabs/wash/api/rql"
+	"github.com/puppetlabs/wash/api/rql/ast/asttest"
+	"github.com/stretchr/testify/suite"
+)
+
+// This file contains real-world meta primary test cases
+//
+// TODO: Once EvalEntrySchema's implemented, add schema predicate
+// tests
+
+type MetaPrimaryRealWorldTestSuite struct {
+	asttest.Suite
+	e rql.Entry
+}
+
+// TTC => TrueTestCase
+func (s *MetaPrimaryRealWorldTestSuite) TTC(key interface{}, predicate interface{}) {
+	q := Query()
+	s.MUM(q, s.A("meta", s.A("object", s.A(s.A("key", key), predicate))))
+	s.Suite.EETTC(q, s.e)
+}
+
+// FTC => FalseTestCase
+func (s *MetaPrimaryRealWorldTestSuite) FTC(key interface{}, predicate interface{}) {
+	q := Query()
+	s.MUM(q, s.A("meta", s.A("object", s.A(s.A("key", key), predicate))))
+	s.Suite.EEFTC(q, s.e)
+}
+
+func (s *MetaPrimaryRealWorldTestSuite) TestMetaPrimary() {
+	s.TTC("architecture", s.A("string", s.A("=", "x86_64")))
+	s.TTC("architecture",
+		s.A("string",
+			s.A("AND",
+				s.A("=", "x86_64"),
+				s.A("glob", "x86*"),
+			),
+		),
+	)
+
+	s.TTC("amiLaunchIndex", s.A("OR", nil, s.A("NOT", nil)))
```

Comment:
So, nil or not nil? That should always pass no matter what's in the file, right?

Reply:
Yeah, this is really testing existence. I'll add a comment about that once the schema stuff's done.